src/librustc_mir/transform/promote_consts.rs

```diff
@@ -322,6 +325,17 @@ impl<'a, 'tcx> Promoter<'a, 'tcx> {
                         _ => bug!()
                     }
                 }
+                Candidate::Repeat(loc) => {
+                    let ref mut statement = blocks[loc.block].statements[loc.statement_index];
+                    match statement.kind {
+                        StatementKind::Assign(_, box Rvalue::Repeat(ref mut operand, _)) => {
+                            let ty = operand.ty(local_decls, self.tcx);
+                            let span = statement.source_info.span;
+                            mem::replace(operand, Operand::Copy(promoted_place(ty, span)))
+                        }
+                        _ => bug!()
```

Comment:
```suggestion
                        _ => bug!("something something fill me in")
```

Reply:
This is pre-existing, and it's a shorter `unreachable!`, in that it should be impossible to hit this (i.e. the `Candidate` is bogus).